nodestream/pipeline/extractors/files.py

```diff
@@ -162,6 +163,16 @@ def get_files(self) -> AsyncIterator[ReadableFile]:
         """
         raise NotImplementedError
 
+    def describe(self) -> str:
+        """Return a human-readable description of the file source.
+
+        This method should return a human-readable description of the file source.
+        The description should be a string that describes the file source in a
+        way that is understandable to the user. The description should be
+        concise and informative.
+        """
+        return str(self)
```

Comment:
Maybe this can be a raise NotImplementedError, unless it's expected that __repr__ will be implemented for a subclass.

Reply:
The thinking here is that it provides a "safe" default. I don't want the integration interface to be too large.

Reply:
Ah that makes sense, I only just noticed that this is just a description of the file, so returning the subclass name by default works out too.

Reply:
Oh there is still a missing unit test here that would encapsulate that case.

Reply:
Added!